metron-platform/metron-enrichment/src/main/java/org/apache/metron/enrichment/bolt/GenericEnrichmentBolt.java

```diff
@@ -149,9 +154,10 @@ public JSONObject load(CacheKey key) throws Exception {
     cache = CacheBuilder.newBuilder().maximumSize(maxCacheSize)
             .expireAfterWrite(maxTimeRetain, TimeUnit.MINUTES)
             .build(loader);
+    GeoLiteDatabase.INSTANCE.update((String)getConfigurations().getGlobalConfig().get(GeoLiteDatabase.GEO_HDFS_FILE));
     boolean success = adapter.initializeAdapter();
```

Comment:
You're right that if we adjust the enrichment adapter to accept configuration values, this can be pushed to the GeoAdapter, where I agree it makes more sense.

Are there concerns over changing the interface method directly, or would we prefer to give `EnrichmentAdapter` a `initializeAdapter(Map<String, Object> config)` with default impl that calls `initializeAdapter()` with no args?  The second option makes it a bit ugly that `GeoAdapter` still has the original `initializeAdapter()`, but has the benefit of keeping the interface backwards compatible to anybody with custom adapters (Is this a thing that people have?).

And you are correct that there is an issue with initializing a missing DB (Great catch!). If we make this change, I believe the only time you'd have an issue is if you set up a geo enrichment and didn't actually set up the DB.  Which is an entirely reasonable issue to have an error with.

@dlyle65535 Would this address your concerns about having the init in the GenericEnrichmentBolt in a satisfactory manner?